src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.h

```diff
@@ -1386,22 +1390,23 @@ ep_rt_utf8_to_utf16le_string (
     if (!str)
         return NULL;
 
-    // Shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // Implementation would just use strlen and malloc to make a new buffer, and would then copy the string chars one by one.
-    // Assumes that only ASCII is used for ep_char8_t
-    size_t len_utf8 = strlen(str);
-    ep_char16_t *str_utf16 = reinterpret_cast<ep_char16_t *>(malloc ((len_utf8 + 1) * sizeof (ep_char16_t)));
-    if (!str_utf16)
+    int32_t flags = MINIPAL_MB_NO_REPLACE_INVALID_CHARS;
+
+    ep_char16_t* lpDestStr = NULL;
+
+    if (static_cast<int>(len) < 0)
+        len = (size_t)strlen(str) + 1;
+
+    size_t ret = (size_t)minipal_get_length_utf8_to_utf16 (str, len, flags);
```

Comment:
```suggestion
    size_t ret = minipal_get_length_utf8_to_utf16 (str, len, flags);
```
`minipal_get_length_utf8_to_utf16`  returns `size_t`. This cast should not be needed.